Xamarin.Forms.Core.UnitTests/ShellModalTests.cs

```diff
@@ -345,6 +345,22 @@ public async Task NavigatingAndNavigatedFiresForShellModal()
 
 		}
 
+		[Test]
+		public async Task GetCurrentPageInModalNavigation()
+		{
+			Shell shell = new Shell();
+			shell.Items.Add(CreateShellItem(shellItemRoute: "NewRoute", shellSectionRoute: "Section", shellContentRoute: "Content"));
+
+			shell.Navigated += (_, __) =>
+			{
+				var page = shell.CurrentPage;
+				Assert.IsNotNull(page);
+				Assert.AreEqual(page.GetType(), typeof(ModalTestPage));
+			};
+
+			await shell.GoToAsync("ModalTestPage");
```

Comment:
Can you move the asserts below the GoToAsync? If Navigated stops firing (for whatever reason) this test will still pass.  You can probably just store CurrentPage to a local variable and then move the asserts below gotoasync